text/0005-media-webrtc-sdk.md

```diff
@@ -0,0 +1,708 @@
+
+Typically, the client will need to perform various actions with the media server, such as:
+
+- Creating/Releasing senders
+- Creating/Releasing receivers
+- Sender actions: pause, resume, switch stream
+- Receiver actions: pause, resume, switch remote source, update priority, and layers
+
+All actions that involve changing tracks will be performed locally first, and then the `update_sdp` command will be sent to the server.
+
+### 4.6.1 Update SDP
+
+Each time we make changes to the WebRTC connection or negotiationneeded event fired, we need to send an `update_sdp` request to the server over the data channel. This request is described below:
+
+**_Cmd:_**: `peer.update_sdp`
+
+**_Data:_**
+
+```
+{
+    sdp: String,
+    tracks: {
+        receivers: [
+            {
+                kind: "audio" | "video",
+                id: String,
+            }
+        ],
+        senders: [
+            {
+                kind: "audio" | "video",
+                id: String,
+                uuid: String,
+                metadata: Option<String>,
+                state: Option<{
+                    active: bool,
+                    screen: bool,
+                }>,
+            }
+        ],
+    }
+}
+```
+
+**_Response data_**:
+
+```
+{
+    sdp: String
+}
+```
+
+### 4.6.2 Room actions, event
+
+We can subscribe to peers event (joined, left, track added, track removed) and also can unsubscribe from it.
+
+#### 4.6.2.1 Subscribe to other peers event
+
+(Note that this action only works with `event.subscribe` manual mode.)
+
+**_Cmd:_** `room.peers.subscribe`
+
+**_Request Data:_**
+
+```
+{
+    peer_ids: [String],
+}
+```
+
+**_Response Data:_** None
+
+#### 4.6.2.2 Unsubscribe to other peers event
+
+(Note that this action only works with `subscribe` manual mode.)
+
+**_Cmd:_**: `room.peers.unsubscribe`
+
+**_Request Data:_**
+
+```
+{
+    peer_ids: [String],
+}
+
+```
+
+#### 4.6.2.3 Peer joined event
+
+**_Cmd:_**: `room.peers.{peer}.joined`
+
+**_Event data:_**:
+
+```
+{
+    metadata: Option<String>,
+}
+```
+
+**_Response Data:_**: None
+
+#### 4.6.2.3 Peer left event
+
+**_Cmd:_**: `room.peers.{peer}.left`
+
+**_Event data:_**: None
+
+**_Response Data:_**: None
+
+#### 4.6.2.3 Track added event
+
+**_Cmd:_**: `room.peers.{peer}.tracks.{track}.added`
+
+**_Event data:_**:
+
+```
+{
+    metadata: Option<String>,
+    state: {
+        active: bool,
+        screen: bool,
+        simulcast: bool,
+    },
+}
+```
+
+#### 4.6.2.3 Track updated event
+
+**_Cmd:_**: `room.peers.{peer}.tracks.{track}.updated`
+
+**_Event data:_**:
+
+```
+{
+    state: {
+        active: bool,
+    },
+}
+```
+
+#### 4.6.2.3 Track removed event
+
+**_Cmd:_**: `room.peers.{peer}.tracks.{track}.removed`
+
+**_Event data:_**: None
+
+**_Response Data:_**: None
+
+### 4.6.3 Session actions, event
+
+#### 4.6.3.1 Disconnect
+
+**_Cmd:_**: `session.disconnect`
+
+**_Request data:_**: None
+
+**_Response:_**: None
+
+#### 4.6.3.2 Goaway event
```

Comment:
should use terms like: down, close or terminated for more clarity

Reply:
I not agree with this, goaway is just server don't need to serve the session anymore, it can be server is scheduled restart, destroy or resource not enough. The details reason will be contain in event body